### ISSUE TITLE
Spike: Create Metadata Validator

### ISSUE THREAD

**Issue by jillpe** [scientist-softserv/west-virginia-university#164]
## Summary
SOW: SoftServ will expand existing Bulkrax functionality to add additional validations for metadata submission by CSV.  Specific requirements will be defined prior to work, but known elements for this validation include:
- Inclusion of required fields
- Standardized spelling and capitalization
- Proper use of delimiters
- Controlled vocabulary terms
This will be designed to allow external partners to validate files without the ability to import.  It will involve adding a permissions layer to Bulkrax so that test imports can be run by users who do not have the permission to import materials.

[Notes from Meeting 2024-12-06](https://docs.google.com/document/d/1MbNb1OIC6jhZJzCM58DiIlalxPKMKm8--1EZhGDvpK0/edit?tab=t.0)
- want their partners to be able to send this into a validator before it gets to an importer
  - they do not enter the site to upload records, they give their csv's to Sarah and she uploads them
- can be anywhere. Does not have to be in bulkrax
- stretch goal would be incorporating into bulkrax so that once it is ready, Sarah can send it to bulkrax
- Inspiration from UVA
  - https://github.com/uvalib/dplava-validation-service 
  - https://github.com/dplava 

[Other Notes](https://docs.google.com/spreadsheets/d/11-tjj3bisDMExE2aOzJr3lFLFRORPW_0h15VhTd7s50/edit?gid=391330904#gid=391330904)
Validator: We are looking at something similar to the DPLA validator, where a partner uploads the xml for their metadata to the DPLA git. Once uploaded to the git, the validator confirms or denies the metadata submission according to required criteria. In our case it might look something like:

A partner uploads metadata missing a required metadata field and the validator denies the submission

We aren’t looking to incorporate the DPLA git validator, just build one like it. It could work with xml or a csv, no real preference there. Ideally the validator would check for things like incorrect spelling, capitalization, and proper use of delimiters, as well as against a controlled vocabulary to ensure terms are correct.

Rob:  This is actually just an expansion of existing Bulkrax validations.

## Acceptance Criteria
- [ ] Follow up tickets for the implementation are created

Look into setting it as a background job with email notification
- [x] #171

**Comment by jillpe** [comment]
Questions to think about:
- Where do we want to implement this?

**Comment by laritakr** [comment]
Will be looking into setting it as a background job with email notification on a new ticket.
- [ ] Validate as a background job
